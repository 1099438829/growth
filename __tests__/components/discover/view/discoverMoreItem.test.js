import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import DiscoverMoreItem from '../../../../src/components/discover/view/DiscoverMoreItem';

const home3 = require('../../../../assets/growth-ui/img/home-3.jpg');

it('renders correctly', () => {
  const moreItem = renderer.create(
    <DiscoverMoreItem
      image={home3}
      title="你还需要什么功能？"
      titleBackgroundColor="rgba(255,255,255, .5)"
      onclick={() => {}}
    />,
  );
  const moreItemJson = moreItem.toJSON();
  expect(moreItemJson).toMatchSnapshot();
});
