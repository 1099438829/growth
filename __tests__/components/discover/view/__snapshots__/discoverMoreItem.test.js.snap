// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "flex": 1,
        "marginTop": 0,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "flex": 7,
          "resizeMode": "stretch",
          "width": 720,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255,255,255, .5)",
            "flex": 3,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#03a9f4",
            }
          }
        >
          你还需要什么功能？
        </Text>
      </View>
    </Image>
  </View>
</View>
`;
